sound/soc/sof/pcm.c

```diff
@@ -254,13 +254,14 @@ static int sof_pcm_hw_free(struct snd_pcm_substream *substream)
 	if (!spcm)
 		return -EINVAL;
 
-	if (!spcm->prepared[substream->stream])
-		return 0;
-
 	dev_dbg(sdev->dev, "pcm: free stream %d dir %d\n", spcm->pcm.pcm_id,
 		substream->stream);
 
-	ret = sof_pcm_dsp_pcm_free(substream, sdev, spcm);
+	if (spcm->prepared[substream->stream]) {
```

Comment:
the dev_dbg() should go in the if() block, no?

Reply:
@plbossart Considered, but I think it belongs outside. The "if (prepared)" block is only an optional  path to send free-pcm-params to DSP if it not was sent after DSP was stopped.

The dev_dbg() applies to whole function and is telling us that the PCM is freed.